toolchain/check/return.cpp

```diff
@@ -42,7 +42,7 @@ static auto NoteReturnType(Context::DiagnosticBuilder& diag,
                            SemIR::TypeId return_type_id) {
   CARBON_DIAGNOSTIC(ReturnTypeHereNote, Note, "return type of function is {0}",
                     SemIR::TypeId);
-  diag.Note(function.return_storage_id, ReturnTypeHereNote, return_type_id);
+  diag.Note(function.return_slot_id, ReturnTypeHereNote, return_type_id);
```

Comment:
Here is an example where it would be better to use an instruction id instead of a TypeId, so that we could match the spelling of the type the user wrote. We have been using TypeIds previously, but the goal is to switch.

Reply:
OK, given an `InstId` representing the type expression the user wrote, how do I obtain the spelling the user wrote? Is there an existing example of the new pattern that I can follow to update this code?

Reply:
Use the type `InstIdAsType` as the type of the diagnostic argument; search for that type name for examples.

Reply:
OK, done.